tests/init/neg/InteractiveDriver.check

```diff
@@ -0,0 +1,5 @@
+-- Error: tests/init/neg/InteractiveDriver.scala:22:71 -----------------------------------------------------------------
+22 |    val l2: Seq[C[?]] = l.collect{ case x: InteractiveDriver[?] => h(x) } // error
+   |                                                                       ^
+   |Calling the external method constructor $anon may cause initialization errors. Calling trace:
+   | -> val l2: Seq[C[?]] = l.collect{ case x: InteractiveDriver[?] => h(x) } // error	[ InteractiveDriver.scala:22 ]
```

Comment:
I see why we have a warning here: we synthesize an anonymous local class here, but its `defTree` is not set. Same for the above.